pkg/fs/real.go

```diff
@@ -54,6 +55,10 @@ func (realFilesystem) Exists(path string) bool {
 	return !os.IsNotExist(err)
 }
 
+func (realFilesystem) ReadDir(dirname string) ([]os.FileInfo, error) {
+	return ioutil.ReadDir(dirname)
+}
```

Comment:
I noticed you used `Walk` everywhere. Because of the annoying `os.File.Readdir{,names}` semantics?
Went with the `ioutil.ReadDir` signature instead unless we strictly want to match `os.File`.